variables.tf

```diff
@@ -167,3 +167,27 @@ variable "mc_extra_commands" {
   default     = [""]
   description = "Extra commands to run in MC container after creating buckets"
 }
+
+# KMS related
+variable "use_vault_kms" {
+  type = bool
+  default = false
+  description = "Use vault transit encryption engine as KMS for transparent encryption (auto-encrypt)"
+}
+
+variable "vault_address" {
+  type = string
+  default = "vault.service.consul:8200"
+  description = "Address to vault service. Only relevant when Vault KMS is used."
+}
+variable "vault_kms_approle_kv" {
+  type = string
+  default = ""
+  description = "Path to key in vault where ApproleID and SecretID is stored. Only relevant when Vault KMS is used."
+}
+
+variable "vault_kms_key_name" {
+  type = string
+  default = ""
+  description = "Name of key in vault transit engine. Only relevant when Vault KMS is used."
+}
```

Comment:
I think it's better to wrap these in a nested variable. Similar to the `vault_secret` variable. WDYT?

Reply:
Yeah I can do that, it´s not that many variables, but if you think it´ll look cleaner with them wrapped that´s fine.

Reply:
I think it makes sense to collect them in one variable because since all should be used in the same context.

Reply:
It did not like variables to be nested when there is a empty default string, so leaving it as is.

Reply:
I don't see the problem with a nested variable. Can you ellaborate?

Reply:
bump